server/container_create.go

```diff
@@ -388,34 +388,14 @@ func buildOCIProcessArgs(containerKubeConfig *pb.ContainerConfig, imageOCIConfig
 	kubeArgs := containerKubeConfig.Args
 
 	// merge image config and kube config
-	// same as docker does today...
-	if imageOCIConfig != nil {
-		if len(kubeCommands) == 0 {
-			if len(kubeArgs) == 0 {
-				kubeArgs = imageOCIConfig.Config.Cmd
-			}
-			if kubeCommands == nil {
-				kubeCommands = imageOCIConfig.Config.Entrypoint
-			}
+	if ociConfig != nil && len(kubeCommands) == 0 {
+		kubeCommands = ociConfig.Entrypoint
+		if len(kubeArgs) == 0 {
+			kubeArgs = ociConfig.Cmd
 		}
 	}
 
-	if len(kubeCommands) == 0 && len(kubeArgs) == 0 {
```

Comment:
We would want to retain this error.

Reply:
> We would want to retain this error.

Can you be more specific about what you see failling?  9574d0e1 is entirely about removing this requirement, which is only important if the container will eventually be `start`ed (see the spec [here][1] and [here][2]).  I'm not very familiar with Kubernetes, but my impression is that starting containers is an explicit separate step from creating them.  And having start-less containers will allow us to close #91 by using the runtime's dummy process for the no-process-image case.

[1]: https://github.com/opencontainers/runtime-spec/blame/v1.0.1/config.md#L145
[2]: https://github.com/opencontainers/runtime-spec/blame/v1.0.1/runtime.md#L119

Reply:
I think we want to fail fast here if we can't find a command. k8s has a separate create/start but the intent is to always start a container so there isn't really a use case for creating a container and not starting it.

Reply:
> k8s has a separate create/start but the intent is to always start a container…

Ok.  I've pushed f41bf468 → a93e7434, which rebases onto master, drops 9574d0e, and shifts the no-args error (so `RunPodSandbox` can have an unset `process`).